synfig-studio/src/gui/dialogs/dialog_setup.cpp

```diff
@@ -784,13 +788,43 @@ Dialog_Setup::create_interface_page(PageInfo pi)
 	{
 	FileSystem::FileList files;
 	FileSystemNative::instance()->directory_scan(ResourceHelper::get_themes_path(), files);
-	for (const auto& dir : files)
-		icon_theme_combo.append(dir);
-	icon_theme_combo.set_active_text(App::get_icon_theme_name());
+	for (const auto& dir : files) {
+		std::string theme_name = dir;
+
+		std::string full_filename = ResourceHelper::get_themes_path() + "/" + dir + "/index.theme";
+
+		if (!FileSystemNative::instance()->is_file(full_filename))
+			continue;
+
+		try {
+			Glib::KeyFile theme_file;
+			theme_file.load_from_file(full_filename);
+			if (theme_file.has_group("Icon Theme")) {
+				if (theme_file.has_key("Icon Theme", "Hidden")) {
```

Comment:
Why not just use something like?
```
if (theme_file.get_boolean("Icon Theme", "Hidden")) {
	continue;
}
```

Reply:
Because [the documentation says it can raise exception](https://developer-old.gnome.org/glibmm/stable/classGlib_1_1KeyFile.html#a0d383ad579647505ff3376e93591aaa7).
https://gitlab.gnome.org/GNOME/glibmm/-/blob/glibmm-2-58/glib/src/keyfile.ccg#L173-L183

Reply:
Lol. Got it.
Should I squash it? Or rebase it please.

Reply:
It's okay to squash it :)